presto-spark-base/src/main/java/com/facebook/presto/spark/execution/http/PrestoSparkHttpWorkerClient.java

```diff
@@ -199,8 +220,79 @@ public ListenableFuture<BaseResponse<TaskInfo>> updateTask(
                 createAdaptingJsonResponseHandler(taskInfoCodec));
     }
 
+    public ListenableFuture<BaseResponse<ServerInfo>> getServerInfo()
+    {
+        Request request = setContentTypeHeaders(false, prepareGet())
+                .setUri(serverUri)
+                .build();
+        return httpClient.executeAsync(request, createAdaptingJsonResponseHandler(serverInfoCodec));
+    }
+
+    public SettableFuture<ServerInfo> getServerInfoWithRetry()
+    {
+        SettableFuture<ServerInfo> future = SettableFuture.create();
+        doGetServerInfo(future);
+        return future;
+    }
+
     public URI getLocation()
     {
         return location;
     }
+
+    private void doGetServerInfo(SettableFuture<ServerInfo> future)
```

Comment:
The retry logic should not be inside of http client class. Can we move it to the caller class? Or somewhere better